src/AppBundle/Controller/Admin/BlogController.php

```diff
@@ -147,6 +148,7 @@ public function editAction(Post $post, Request $request)
         if ($form->isSubmitted() && $form->isValid()) {
             $post->setSlug($this->get('slugger')->slugify($post->getTitle()));
             $entityManager->flush();
+            $this->getDoctrine()->getRepository(Tag::class)->cleanUnusedTags();
```

Comment:
as you do it using a separate query, you should wrap the flush and this query in a transaction, to be atomic.